### DISCUSSION TITLE
Is it ok that that in AsyncNotifier excpetions aren't automatically caught during development??

### DISCUSSION THREAD

**Discussion by AlbertoNoris** [rrousselGit/riverpod#1896] [Q&A] [answered]
Hi!
Is it ok that in this piece of code:
```
class ProvaNotifier extends AsyncNotifier<int> {
  @override
  FutureOr<int> build() async {
    return ListOfHttpCalls.function();
  }
}

class ListOfHttpCalls {
  static Future<int> function() async {
    await Future.delayed(Duration(seconds: 1));
    throw Exception();
  }
}
```
the execution stops on the Exception inside `ListOfHttpCalls.function()` ?
![image](https://user-images.githubusercontent.com/61833464/201382631-50e23aed-adac-4ad0-b6e1-8ca598b109fd.png)

Reading the documentation of the [build ](https://pub.dev/documentation/riverpod/latest/riverpod/AsyncNotifier/build.html)method this is something that could very well happen so shouldn't it gloss over it like `return ListOfHttpCalls.function();` was inside a try catch block?

**Comment by rrousselGit** [comment, marked as answer]
The exception is caught and the UI will show an error (if using `when`).  
At the same time Riverpod voluntarily adds the necessary for the IDE to detect this exception to make debugging easier. 

It's voluntary that the IDE pauses when you have an error

**Reply by AlbertoNoris** [reply]
Thanks!
